src/main.js

```diff
@@ -297,12 +297,14 @@ const CodingManagerApplication = new Lang.Class({
         let active_window_xid = active_window ? active_window.get_xid() : 0;
         let current_window_xid = current_window ? current_window.get_xid() : 0;
 
-        // try to match transient windows
-        let transient_window = active_window.get_transient();
+        if (active_window != null) {
```

Comment:
I would actually early return here. Also != should be !==. I restructured that function a bit, my suggestion would be:

```
diff --git a/src/main.js b/src/main.js
index 63a102f..943378e 100644
--- a/src/main.js
+++ b/src/main.js
@@ -294,22 +294,19 @@ const CodingManagerApplication = new Lang.Class({
     _on_active_window_changed: function() {
         let active_window = Wnck.Screen.get_default().get_active_window();
         let current_window = this._window.get_window();
-        let active_window_xid = active_window ? active_window.get_xid() : 0;
         let current_window_xid = current_window ? current_window.get_xid() : 0;
 
-        if (active_window != null) {
-            // try to match transient windows
-            let transient_window = active_window.get_transient();
+        if (active_window === null)
+            return;
 
-            if (transient_window != null &&
-                current_window_xid == transient_window.get_xid()) {
-                return;
-            }
-        }
+        // try to match transient window
+        let transient_window = active_window.get_transient();
+        if (transient_window !== null &&
+            current_window_xid === transient_window.get_xid())
+            return;
 
-        if (active_window_xid !== current_window_xid) {
+        if (active_window.get_xid() !== current_window_xid)
             this.hide();
-        }
     },
 
     _update_geometry: function() {

```